File: pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;

/**
 * @author lihai
 * @since 2021/2/16 0016
 */
public interface BrandService {
    List<TbBrand> findAll();

    PageResult findBrandByPage(int pageNum, int pageSize);

    PageResult findBrandByPage(TbBrand brand, int pageNum, int pageSize);

    void add(TbBrand brand);

    TbBrand findById(Long id);

    void updateBrand(TbBrand brand);

    void delete(Long[] ids);
}

File: pinyougou_manager_web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {
    this.findAll =function () {
        return $http.get("../brand/findAll.do");
    }
    this.findBrandByPage =function (currentPage,itemsPerPage,searchEntity) {
        return $http.post("../brand/findBrandByPage.do?pageNum="+currentPage+
            "&pageSize="+itemsPerPage, searchEntity);
    }
    this.add =function (entity) {
        return $http.post("../brand/add.do",entity);
    }
    this.update = function (entity) {
        return $http.post("../brand/update.do",entity);
    }
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id=" +id);
    }
    this.delete =function (selectIds) {
        return $http.get('../brand/delete.do?ids='+selectIds);
    }
});